karate-core/src/main/java/com/intuit/karate/core/Config.java

```diff
@@ -333,6 +379,7 @@ public Config(Config parent) {
         responseHeaders = parent.responseHeaders;
         afterScenario = parent.afterScenario;
         afterFeature = parent.afterFeature;
+        // purposely not passing ignoredStepFailures to the children config
```

Comment:
I think in all other config - we do cascade and propagate to child config. what's the rationale here ?

Reply:
Rationale is reflected on this comment: https://github.com/intuit/karate/issues/1446#issuecomment-765597285

I think this feature is necessary to get as much information as possible whenever a test fails (and so capture all the failed matches and not only the first) but I also recognize it's opening the door for some bad practices. For that reason there were two concepts I kept in mind:

- its usage need to be explicit and so if the desire is to reuse across multiple feature files the testers should add it on those feature files. Allowing it to naturally cascade down to features seems like a recipe for unintended consequences when using the code below;
- this should open the door to be used as an "if" condition for large sections of the tests and so even with the flag to continue the test step execution, once the scenario execution is done it'll still be marked as a failure (also why I renamed to "continueOnStepFailure" instead of ignore)


code snippet for why I prefer to avoid propagating this config:
```
* configure ignoreStepFailure = true
* match var == 'foo'
* call read('reusable-feature-with-some-matches-that-does-something-else-that-tester-is-unaware.feature')
* match var == 'bar'
* configure ignoreStepFailure = false

```

Reply:
I'm fine with unintended consequences.

a) this is a feature I don't recommend anyway
b) ideally feature re-use should not be for assertions

my vote is for consistency with the rest of karate. we can change this in future if it is really a problem